internal/synclet/client_adapter.go

```diff
@@ -69,6 +71,14 @@ func (s *SyncletCli) UpdateContainer(
 	for {
 		reply, err := stream.Recv()
 
+		if reply != nil && reply.FailedRunStep != nil {
+			frs := reply.FailedRunStep
+			return build.RunStepFailure{
```

Comment:
this is going to leak a goroutine. from the GRPC docs:

```
NewStream creates a new Stream for the client side. This is typically called by generated code. ctx is used for the lifetime of the stream.

To ensure resources are not leaked due to the stream returned, one of the following actions must be performed:

1. Call Close on the ClientConn.
2. Cancel the context provided.
3. Call RecvMsg until a non-nil error is returned. A protobuf-generated
   client-streaming RPC, for instance, might use the helper function
   CloseAndRecv (note that CloseSend does not Recv, therefore is not
   guaranteed to release all resources).
4. Receive a non-nil, non-io.EOF error from Header or SendMsg.
```

Reply:
(probably best thing to do is save this in a variable, and only return it after the EOF)

Reply:
went with cancelling the context instead because it seemed potentially gnarly to rely on getting an EOF--what if something DIFFERENT went wrong with the stream? I assume we'd still want to return the `RunStepFailure` error...) -- but lmk if this is wrong and i will go the storing-on-a-variable route

Reply:
...nvm it was giving me lint errors, I just went with your suggestion.